api/utils/auth.js

```diff
@@ -0,0 +1,27 @@
+const jwt = require('jsonwebtoken');
+const { userService } = require('../services');
+
+const { asyncWrap } = require('./error');
+
+const loginRequired = asyncWrap(async (req, res, next) => {
+    const accessToken = req.headers.authorization;
+
+    if (!accessToken) {
+        const error = new Error('Need access token');
+        error.statucCode = 401;
+        throw error;
+    }
+
+    const payload = await jwt.verify(accessToken, process.env.JWT_SECRET);
+    const user = await userService.getUserById(payload.id);
+
+    if (!user) {
+        const error = new Error('User not found');
+        error.statucCode = 404;
+        throw error;
```

Comment:
사용자에 대한 데이터가 없기 때문에 404 not found를 사용한 것으로 보입니다. 만약 제품이나 장바구니 같은 데이터 였다면 괜찮았을 것 같은데, 사용자 데이터는 결국 권한과 연관 있는 데이터로 볼 수 있습니다.
따라서 권한과 관련된 status code로 401을 사용하시면 좋을 것 같습니다.
 
```suggestion
        const error = new Error('Invalid user');
        error.statucCode = 401;
        throw error;
```